### PR TITLE
PROXY-7: return transformed data

### PR DESCRIPTION
**Pull request by i-siz**
- /meteors returns transformed asteriod data with the following properties: id, name, average_diameter_in_meters, is_potentially_hazardous_asteroid, close_approach_date_full, relative_velocity_in_kilometers_per_second.
- all asteroid evidences are grupped and sorted by date.
- moment library is replaced with date-fns library

### CODE REVIEW

**Comment by mdziarkach** [comment]
please create `app` or `src` folder  and transfer all your folders to it

**Comment by mdziarkach** [comment]
please create `app` or `src` folder  and transfer all your folders to it

File: controllers/asteroids [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const getAsteroidsWithinPeriod = require('../services/asteroidsData');
```
Comment (mdziarkach) [inline controllers/asteroids]: please create `app` or `src` folder  and transfer all your folders to it

File: transformers/asteroids [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const { getAverageValue } = require('../helpers/math');
+
+const transformAsteroidsData = (asteroidsData) => ({
```
Comment (mdziarkach) [inline transformers/asteroids]: this is not critical, but such transformations are usually called mappers

File: controllers/asteroids [hunk 3]
```diff
@@ -8,7 +9,8 @@ const handleError = (res, error) => {
 const getAsteroidsData = async (req, res) => {
     try {
         const asteroidsData = await getAsteroidsWithinPeriod(startDate, endDate);
-        res.status(200).json(asteroidsData);
+        const transformedData = transformAsteroidsData(asteroidsData);
+        res.status(200).json(transformedData);
```
Comment (mdziarkach) [inline controllers/asteroids]: express returns status 200 by default
https://stackoverflow.com/questions/53403078/express-is-it-necessary-to-respond-with-status-200

I assume that it is better to make the responses of the same type and return, for example, `data`
```js
const data= transformAsteroidsData(asteroidsData);
res.json({ data });
```

File: helpers/dates.js [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+const DATE_TEMPLATE = 'yyyy-MM-dd';
 
-const lastFriday = moment().day(-2);
-const mondayPrecedingLastFriday = moment().day(-2).day(1);
-const startDate = mondayPrecedingLastFriday.format(DATE_TEMPLATE);
-const endDate = lastFriday.format(DATE_TEMPLATE);
+const lastFriday = previousFriday(new Date());
+const mondayPrecedingLastFriday = previousMonday(lastFriday);
+const startDate = format(mondayPrecedingLastFriday, DATE_TEMPLATE);
+const endDate = format(lastFriday, DATE_TEMPLATE);
```
Comment (mdziarkach) [inline helpers/dates.js]: these values do not look like utilities or helpers. They are used only once in the code and cannot be reused
Maybe it makes sense to move them to the piece of code where you need them?

File: transformers/asteroids [hunk 5]
```diff
@@ -0,0 +1,23 @@
+const { getAverageValue } = require('../helpers/math');
```
Comment (mdziarkach) [inline transformers/asteroids]: you forgot about the file extension
it works?

File: controllers/asteroids (hunk 1 above)
Comment (i-siz) [reply controllers/asteroids]: Should the entry point file server.js be moved to this folder too?

File: src/routes/meteors.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 const express = require('express');
-const { getAsteroidsData } = require('../controllers/asteroids');
+const { getAsteroidsData } = require('../controllers/asteroids.js');
```
Comment (plutsenka) [inline src/routes/meteors.js]: Seems like ../controllers/asteroids should be enough (here & below)

**Review by plutsenka — APPROVED** [review]

File: src/routes/meteors.js (hunk 6 above)
Comment (i-siz) [reply src/routes/meteors.js]: Thank you. Updated.

File: controllers/asteroids (hunk 1 above)
Comment (i-siz) [reply controllers/asteroids]: Thank you. src folder is created and all folders moved to it.

Comment (mdziarkach) [reply controllers/asteroids]: yes please

File: transformers/asteroids (hunk 2 above)
Comment (i-siz) [reply transformers/asteroids]: Thanks. Remaned.

File: controllers/asteroids (hunk 3 above)
Comment (i-siz) [reply controllers/asteroids]: Updated.

File: controllers/asteroids (hunk 1 above)
Comment (i-siz) [reply controllers/asteroids]: Done.

File: helpers/dates.js (hunk 4 above)
Comment (i-siz) [reply helpers/dates.js]: Updated.

File: transformers/asteroids (hunk 5 above)
Comment (i-siz) [reply transformers/asteroids]: Yes. It works. But I really forgot to add .js extension some files. Now I've fixed it.

**Pull request merged**